services/web/server/src/simcore_service_webserver/exporter/formatters/sds/text_files.py

```diff
@@ -0,0 +1,75 @@
+from typing import Union
+from pathlib import Path
+
+
+class BaseTextFile:
+    def _check_attribute(self, attribute_name: str):
+        if getattr(self, attribute_name) is None:
+            raise ValueError(f"'{attribute_name}' attribute is None, please define it")
+
+    def __init__(self, path: Union[str, Path] = None, text: str = None):
```

Comment:
Here you have a class per text file. Nice. It has two attributes and you want to make sure that the "mandatory" (i.e. required) are set. 
Why not using pydantic or ``dataclass`` classes??  These features are totally suported... and probably more extendable for future changes!

Reply:
though it would look strange with pydantic, but it feels ok. changed it